src/Dependencies/Collections/README.md

```diff
@@ -0,0 +1,5 @@
+﻿# Microsoft.CodeAnalysis.Collections
+
+This package contains shared code implementing specialized collection types.
+
+* `SegmentedArray<T>`: This type behaves similarly to `T[]`, but segments the underlying storage to avoid allocating arrays in the Large Object Heap.

```

Comment:
Is there any good advice on when to use one over the other that you could include / link to here ?

Reply:
The long form might look something like this:

> If your code relies on the indexer performance characteristics of `T[]`, does not rely on `AsSpan()`/`AsMemory()`, and performance measurements show problematic results from Large Object Heap allocations, it may benefit from using `SegmentedArray<T>` instead of `T[]`.

It seems likely that once a user knows that Large Object Heap is a problem that needs to be avoided, the statement already in the README would be sufficient to help them in determining the correct type to use.

All the types I currently have planned for this package are highly application-specific and motivated by profiling results.

Reply:
> All the types I currently have planned for this package are highly application-specific and motivated by profiling results.

Seems like a good bit to include. For example linking to a PR or two where you made conversions to `SegmentedArray<T>` due to profiling pushing you in that direction. Particularly if the PR description has data in it then it would serve as a nice "treasure map" for other developers to follow. 

> It seems likely that once a user knows that Large Object Heap is a problem that needs to be avoided, the statement already in the README would be sufficient to help them in determining the correct type to use.

Think you're applying too much of your internal understanding of performance and this type to the answer here. Think a good chunk of devs would find this type who don't quite have the understanding level you're expecting here.

Reply:
👍 Added some details